ivadomed/scripts/prepare_dataset.py

```diff
@@ -0,0 +1,146 @@
+        elif aim == 'c2':
+            if arr[x, y, z] == 3:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+    list_label_image.sort(key=lambda x: x[3])
+    return list_label_image
+
+
+def extract_mid_slice_and_convert_coordinates_to_heatmaps(bids_path, suffix, aim="full"):
+    """
+    This function takes as input a path to a dataset  and generates a set of images:
+   (i) mid-sagittal image and
+   (ii) heatmap of disc labels associated with the mid-sagittal image.
+
+    Args:
+        bids_path (string): path to BIDS dataset form which images will be generated
+        suffix (string): suffix of image that will be processed (e.g., T2w)
+        aim (string): 'full' or 'c2'. If 'c2' retrieves only c2 label (value = 3) else create heatmap with all label.
+    Returns:
+        None. Images are saved in BIDS folder
+    """
+    t = os.listdir(bids_path)
+    t.remove('derivatives')
+
+    for i in range(len(t)):
+        sub = t[i]
+        path_image = os.path.join(bids_path, t[i], 'anat', t[i] + suffix + '.nii.gz')
+        if os.path.isfile(path_image):
+            path_label = os.path.join(bids_path, 'derivatives', 'labels', t[i], 'anat', t[i] + suffix +
+                                      '_label-disc-manual.nii.gz')
+            list_points = mask2label(path_label, aim=aim)
+            image_ref = nib.load(path_image)
+            nib_ref_can = nib.as_closest_canonical(image_ref)
+            imsh = np.array(nib_ref_can.dataobj).shape
+            mid_nifti = imed_preprocessing.get_midslice_average(path_image, list_points[0][0], slice_axis=0)
+            nib.save(mid_nifti, os.path.join(bids_path, t[i], 'anat', t[i] + suffix + '_mid.nii.gz'))
+            lab = nib.load(path_label)
+            nib_ref_can = nib.as_closest_canonical(lab)
+            label_array = np.zeros(imsh[1:])
+
+            for j in range (len(list_points)):
+                label_array[list_points[j][1], list_points[j][2]] = 1
+
+            heatmap = heatmap_generation(label_array[:, :], 10)
+            arr_pred_ref_space = imed_utils.reorient_image(np.expand_dims(heatmap[:, :], axis=0), 2, lab, nib_ref_can)
+            nib_pred = nib.Nifti1Image(arr_pred_ref_space, lab.affine)
+            nib.save(nib_pred, os.path.join(bids_path, 'derivatives', 'labels', t[i], 'anat', t[i] + suffix +
+                                            '_mid_heatmap.nii.gz'))
+        else:
+            pass
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--path", dest="path", required=True, type=str,
+                        help="Path to bids folder with " / " at the end.")
+    parser.add_argument("-s", "--suffix", dest="suffix", required=True,
+                        type=str, help="suffix of image file")
+    parser.add_argument("-a", "--aim", dest="aim", default="full", type=str,
+                        help="If set to 'c2' only points with value 3 will be converted to heatmap")
```

Comment:
What about indicating the choices? ie full or c2?
Also, what about being more general: input the label of interest --> 3 for C2/C3, 5 for C4/C5, -1 for all?

Reply:
Yeah I was thinking about it afterward! I'll see what I can do about it !

Reply:
Done in 66a3132